lms/lmsdb/models.py

```diff
@@ -426,6 +446,36 @@ def exercise_number_save_handler(model_class, instance, created):
         instance.number = model_class.get_highest_number() + 1
 
 
+class ExerciseTag(BaseModel):
+    exercise = ForeignKeyField(Exercise)
+    tag = ForeignKeyField(ExerciseTagText)
+    date = DateTimeField(default=datetime.now)
+
+    @classmethod
+    def by_exercise(
+        cls, exercise: Exercise,
+    ) -> Union[Iterable['ExerciseTag'], 'ExerciseTag']:
+        return (
+            cls
+            .select()
+            .where(cls.exercise == exercise)
+            .order_by(cls.date)
+        )
+
+    @classmethod
+    def is_course_tag_exists(cls, course: Course, tag_name: str):
+        return (
+            cls
+            .select()
+            .join(ExerciseTagText)
+            .where(ExerciseTagText.text == tag_name)
+            .switch()
+            .join(Exercise)
+            .where(Exercise.course == course)
+            .exists()
+        )
+
+
```

Comment:
Maybe we should create a bridge of `tags.py` to allow `tags.get_exercises_of('loops')` and `tags.of_exercise(id=10)`/`tags.of_exercise(course=5, number=4)`